### PR TITLE
added tracker response to provide status info

### PR DESCRIPTION
**Pull request by xavluiz**
this adds a tracker response to provide status info to the tracker clients

### CODE REVIEW

File: src/utils/env_helper.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (bjacobson26) [inline src/utils/env_helper.ts:1]: nitpick - can you remove the whitespace up here

File: src/index.ts [hunk 2]
```diff
@@ -23,17 +26,16 @@ swdcTracker.initialize = async (swdcApiHost: string, namespace: string, appId: s
...
     } else {
-      swdcTracker.testMode = true
-      console.log('swdc-tracker test mode on. set env ENABLE_SWDC_TRACKER to "true" to send events')
+      console.log(`swdc-tracker initialized and ready to send events to ${tracker_api_host}`);
     }
-    return { status: "success" };
+
+    return success();
   } catch (e) {
     console.log("swdcTracker failed to initialize", e);
```
Comment (bjacobson26) [inline src/index.ts:37]: do we still need this console log? can probably remove since were throwing an error

File: src/utils/response.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  Success = "success",
+  Failed = "failed",
+}
+
+export class TrackerResponse {
+  status: number = 200;
+  state: TrackerResponseType = TrackerResponseType.Success;
+  mode: TrackerMode = getTrackerMode();
+  message: string = "";
+  data: any = {};
```
Comment (bjacobson26) [inline src/utils/response.ts:13]: what `data` could be returned in a TrackerResponse?

File: src/utils/response.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { getTrackerMode, TrackerMode } from "./env_helper";
+
+export enum TrackerResponseType {
+  Success = "success",
+  Failed = "failed",
+}
+
+export class TrackerResponse {
+  status: number = 200;
+  state: TrackerResponseType = TrackerResponseType.Success;
```
Comment (bjacobson26) [inline src/utils/response.ts:10]: do we need state and status? can we assume 200 status is success (maybe with message set to "success") and anything else is an error?

File: src/index.ts (hunk 2 above)
Comment (xavluiz) [reply src/index.ts:37]: no we don't need it, you're right

File: src/utils/env_helper.ts (hunk 1 above)
Comment (xavluiz) [reply src/utils/env_helper.ts:1]: yep, will do

File: src/utils/response.ts (hunk 3 above)
Comment (xavluiz) [reply src/utils/response.ts:13]: not sure yet. the snowplow-tracker only returns an object that has add, addDict, addJson, and build function, so I don't really know what we can return. i'll take it out until we have a better use case, right?

File: src/utils/response.ts (hunk 4 above)
Comment (xavluiz) [reply src/utils/response.ts:10]: yeah, we only need one or the other. thanks again for asking these questions. i want to tighten this down. which one do you prefer?

File: src/utils/response.ts (hunk 3 above)
Comment (bjacobson26) [reply src/utils/response.ts:13]: sounds good - can leave it out for now

File: src/utils/response.ts (hunk 4 above)
Comment (bjacobson26) [reply src/utils/response.ts:10]: i think status 200 is good

File: src/utils/response.ts (hunk 3 above)
Comment (xavluiz) [reply src/utils/response.ts:13]: actually i'm using data to return the payload properties when in test mode. lets keep it for now and we can figure out what we can return for a prod data response

Comment (xavluiz) [reply src/utils/response.ts:13]: sorry, i think you're right. it's too redundant to have data, i'll remove it

**Comment by xavluiz** [comment]
- removed "data" and "state" from the response wrapper

**Pull request merged**